base/cvd/cuttlefish/host/commands/cvd/instance_manager.cpp

```diff
@@ -241,19 +227,32 @@ cvd::Status InstanceManager::CvdClear(const SharedFD& out,
   std::lock_guard lock(instance_db_mutex_);
   cvd::Status status;
   const std::string config_json_name = cpp_basename(GetGlobalConfigFileLink());
-  auto&& instance_groups = instance_db_.InstanceGroups();
+  auto instance_groups_res = instance_db_.InstanceGroups();
+  if (!instance_groups_res.ok()) {
+    WriteAll(err, fmt::format("Failed to access instance database: {}",
+                              instance_groups_res.error().Message()));
+    status.set_code(cvd::Status::INTERNAL);
+    return status;
+  }
+  auto instance_groups = *instance_groups_res;
   for (const auto& group : instance_groups) {
-    auto config_path = group->GetCuttlefishConfigPath();
+    auto config_path = group.GetCuttlefishConfigPath();
     if (config_path.ok()) {
-      auto stop_result = IssueStopCommand(out, err, *config_path, *group);
+      auto stop_result = IssueStopCommand(out, err, *config_path, group);
       if (!stop_result.ok()) {
         LOG(ERROR) << stop_result.error().FormatForEnv();
       }
     }
-    RemoveFile(group->HomeDir() + "/cuttlefish_runtime");
-    RemoveFile(group->HomeDir() + config_json_name);
+    RemoveFile(group.HomeDir() + "/cuttlefish_runtime");
+    RemoveFile(group.HomeDir() + config_json_name);
```

Comment:
Yes, there are many small fixes like that needed all over the code base, I was planning to address them in a future PR, because some of that code is actually going to be deleted by the following two.